src/Catalog/Persistence/Storage.cs

```diff
@@ -93,10 +105,12 @@ public override string ToString()
             }
             catch (Exception e)
             {
-                string message = String.Format("DELETE EXCEPTION: {0} {1}", resourceUri, e.Message);
-                Trace.WriteLine(message);
+                var message = TraceException(nameof(Delete), resourceUri, e);
                 throw new Exception(message, e);
```

Comment:
Can we just rethrow here `throw;`? The stack on the outermost exception will be wrong if we don't rethrow.